L1Trigger/L1TMuonOverlapPhase1/src/Omtf/OMTFinputMaker.cc

```diff
@@ -0,0 +1,454 @@
+
+  MuonStub stub;
+  stub.type = MuonStub::CSC_PHI_ETA;
+  stub.phiHw = angleConverter->getProcessorPhi(
+      OMTFinputMaker::getProcessorPhiZero(config, iProcessor), procTyp, CSCDetId(rawid), digi);
+  stub.etaHw = angleConverter->getGlobalEta(rawid, digi);
+  stub.phiBHw = digi.getPattern();  //TODO change to phiB when implemented
+  stub.qualityHw = digi.getQuality();
+
+  stub.bx = digi.getBX() - config->cscLctCentralBx();  //TODO sholdn't  it be getBX0()?
+  //stub.timing = digi.getTiming(); //TODO what about sub-bx timing, is is available?
+
+  //stub.etaType = ?? TODO
+  stub.logicLayer = iLayer;
+  stub.detId = rawid;
+
+  OMTFinputMaker::addStub(config, muonStubsInLayers, iLayer, iInput, stub);
+  ///Accept CSC digis only up to eta=1.26.
+  ///The nominal OMTF range is up to 1.24, but cutting at 1.24
+  ///kill efficnency at the edge. 1.26 is one eta bin above nominal.
+  //if(abs(iEta)>1.26/2.61*240) continue;
+  //if (abs(iEta) > 115) continue;
+
+  //    std::cout <<" ADDING CSC hit, proc: "<<iProcessor<<" iPhi : " << iPhi <<" iEta: "<< iEta << std::endl;
+}
+
+bool CscDigiToStubsConverterOmtf::acceptDigi(const CSCDetId& csc, unsigned int iProcessor, l1t::tftype procType) {
+  if (procType == l1t::tftype::omtf_pos && (csc.endcap() == 2 || csc.ring() == 1 || csc.station() == 4))
+    return false;
+  if (procType == l1t::tftype::omtf_neg && (csc.endcap() == 1 || csc.ring() == 1 || csc.station() == 4))
+    return false;
+
+  if (procType == l1t::tftype::emtf_pos && (csc.endcap() == 2 || (csc.station() == 1 && csc.ring() == 3)))
+    return false;
+  if (procType == l1t::tftype::emtf_neg && (csc.endcap() == 1 || (csc.station() == 1 && csc.ring() == 3)))
+    return false;
+
+  unsigned int aSector = csc.chamber();
+  unsigned int aMin = config->getEndcap10DegMin()[iProcessor];
+  unsigned int aMax = config->getEndcap10DegMax()[iProcessor];
+
+  if ((procType == l1t::tftype::emtf_pos || procType == l1t::tftype::emtf_neg) && csc.station() > 1 &&
+      csc.ring() == 1) {
+    aMin = config->getEndcap20DegMin()[iProcessor];
+    aMax = config->getEndcap20DegMax()[iProcessor];
+  }
+
+  if (aMax > aMin && aSector >= aMin && aSector <= aMax)
+    return true;
+  if (aMax < aMin && (aSector >= aMin || aSector <= aMax))
+    return true;
+
+  return false;
+}
+
+void RpcDigiToStubsConverterOmtf::addRPCstub(MuonStubPtrs2D& muonStubsInLayers,
+                                             const RPCDetId& roll,
+                                             const RpcCluster& cluster,
+                                             unsigned int iProcessor,
+                                             l1t::tftype procTyp) {
+  //      int iPhiHalfStrip1 = myangleConverter->getProcessorPhi(getProcessorPhiZero(iProcessor), type, roll, cluster.first);
+  //      int iPhiHalfStrip2 = myangleConverter->getProcessorPhi(getProcessorPhiZero(iProcessor), type, roll, cluster.second);
+
```

Comment:
You can delete these two empty lines, to make clear that these definitions are intended for the debug `cout` below